### PR TITLE
rcar_gen3: plat: Prevent PCIe hang during L1X config access

### PR DESCRIPTION
**Pull request by marex**
Small fix adding PCIe link L1 state fixup.

### CODE REVIEW

File: plat/renesas/rcar/platform.mk [hunk 1]
```diff
@@ -13,6 +13,12 @@ GENERATE_COT			:= 1
 BL2_AT_EL3			:= 1
 ENABLE_SVE_FOR_NS		:= 0
 
+CRASH_REPORTING		:=	1
+$(eval $(call add_define,CRASH_REPORTING))
```
Comment (antonio-nino-diaz-arm) [inline plat/renesas/rcar/platform.mk]: You don't need `` $(eval $(call add_define,CRASH_REPORTING))`` or `` $(eval $(call add_define,HANDLE_EA_EL3_FIRST))``. This platform makefile is included before bl31.mk and before the list of assert_boolean and add_define in the Makefile.

File: plat/renesas/rcar/rcar_common.c [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+#define MSTP319			(1 << 19)
+#define PMSR			0x5c
+#define PMSR_L1FAEG		(1 << 31)
+#define PMSR_PMEL1RX		(1 << 23)
+#define PMCTLR			0x60
+#define PMSR_L1IATN		(1 << 31)
+
+static int rcar_pcie_fixup(unsigned int controller)
+{
+	uint32_t rcar_pcie_base[] = { 0xfe011000, 0xee811000 };
```
Comment (antonio-nino-diaz-arm) [inline plat/renesas/rcar/rcar_common.c]: You should probably check that controller is less than 2.

**Comment by antonio-nino-diaz-arm** [comment]
IP Review +1

**Comment by marex** [comment]
On 2/20/19 2:39 PM, Antonio Niño Díaz wrote:
> *@antonio-nino-diaz-arm* commented on this pull request.
> 
> ------------------------------------------------------------------------
> 
> In plat/renesas/rcar/platform.mk
> <https://github.com/ARM-software/arm-trusted-firmware/pull/1833#discussion_r258483064>:
> 
>> @@ -13,6 +13,12 @@ GENERATE_COT			:= 1
>  BL2_AT_EL3			:= 1
>  ENABLE_SVE_FOR_NS		:= 0
>  
> +CRASH_REPORTING		:=	1
> +$(eval $(call add_define,CRASH_REPORTING))
> 
> You don't need |$(eval $(call add_define,CRASH_REPORTING))| or |$(eval
> $(call add_define,HANDLE_EA_EL3_FIRST))|. This platform makefile is
> included before bl31.mk and before the list of assert_boolean and
> add_define in the Makefile.

I need the HANDLE_EA_EL3_FIRST , otherwise the exception won't be routed
into the ATF, but rather into Linux. Or do you mean to just drop the
eval bit ?

> ------------------------------------------------------------------------
> 
> In plat/renesas/rcar/rcar_common.c
> <https://github.com/ARM-software/arm-trusted-firmware/pull/1833#discussion_r258483825>:
> 
>> +
> +#include <lib/mmio.h>
> +
> +#define CPG_BASE		0xE6150000
> +#define CPG_MSTPSR3		0x0048
> +#define MSTP318			(1 << 18)
> +#define MSTP319			(1 << 19)
> +#define PMSR			0x5c
> +#define PMSR_L1FAEG		(1 << 31)
> +#define PMSR_PMEL1RX		(1 << 23)
> +#define PMCTLR			0x60
> +#define PMSR_L1IATN		(1 << 31)
> +
> +static int rcar_pcie_fixup(unsigned int controller)
> +{
> +	uint32_t rcar_pcie_base[] = { 0xfe011000, 0xee811000 };
> 
> You should probably check that controller is less than 2.

That can never happen if you read the code further.

-- 
Best regards,
Marek Vasut

**Comment by antonio-nino-diaz-arm** [comment]
> I need the HANDLE_EA_EL3_FIRST , otherwise the exception won't be routed into the ATF, but rather into Linux. Or do you mean to just drop the eval bit ?

Just the evals.

> That can never happen if you read the code further.

Fair enough.

**Comment by marex** [comment]
On 2/20/19 3:34 PM, Antonio Niño Díaz wrote:
>     I need the HANDLE_EA_EL3_FIRST , otherwise the exception won't be
>     routed into the ATF, but rather into Linux. Or do you mean to just
>     drop the eval bit ?
> 
> Just the evals.

I presume the PR magically updated itself with this change now .

>     That can never happen if you read the code further.
> 
> Fair enough.
[...]

-- 
Best regards,
Marek Vasut

**Comment by antonio-nino-diaz-arm** [comment]
jenkins: test this please

**Pull request merged**